retrier/retrier.go

```diff
@@ -34,9 +35,15 @@ func New(backoff []time.Duration, class Classifier) *Retrier {
 	}
 }
 
+// WithInfiniteRetry set the retrier to loop infinitely on the last backoff duration
```

Comment:
Minor nits:
- missing a final period
- we should probably be explicit that, if you call this, then your program may hang forever waiting for the work function to succeed

Reply:
You're right, I've made the change 👍